mne/viz/raw.py

```diff
@@ -566,31 +566,59 @@ def _set_psd_plot_params(info, proj, picks, ax, area_mode):
             ax_list, make_label)
 
 
-def _convert_psds(psds, dB, scaling, unit, ch_names):
-    """Convert PSDs to dB (if necessary) and appropriate units."""
-    if dB:
-        where = np.where(psds.min(1) <= 0)[0]
-        if len(where) > 0:
-            raise ValueError("Infinite value in PSD for channel(s) %s. "
-                             "These channels might be dead." %
-                             ', '.join(ch_names[ii] for ii in where))
+def _convert_psds(psds, dB, estimate, scaling, unit, ch_names):
+    """Convert PSDs to dB (if necessary) and appropriate units.
+
+    The following table summarizes the relationship between the value of
+    parameters `dB` and `estimate`, and the type of plot and corresponding
+    units.
+
+    | dB    | estimate    | plot | units           |
+    |-------+-------------+------+-----------------|
+    | True  | 'power'     | PSD  | u**2/Hz (dB)    |
+    | False | 'amplitude' | ASD  | u/sqrt{Hz}      |
+    | True  | 'auto'      | PSD  | u**2/Hz (dB)    |
+    | False | 'power'     | PSD  | u**2/Hz         |
+    | True  | 'amplitude' | ASD  | u/sqrt{Hz} (dB) |
+    | False | 'auto'      | ASD  | u/sqrt{Hz}      |
+    """
+
+    where = np.where(psds.min(1) <= 0)[0]
+    if len(where) > 0:
+        raise ValueError("Infinite value in PSD for channel(s) %s. "
+                         "These channels might be dead." %
+                         ', '.join(ch_names[ii] for ii in where))
+
+    if dB and ((estimate == 'power') or (estimate == 'auto')):
         psds *= scaling * scaling
         np.log10(psds, out=psds)
         psds *= 10
-        ylabel = '%s/Hz (dB)' % unit
-    else:
+        ylabel = '%s**2/Hz (dB)' % unit # Better to use latex and ^?
+
+    if not dB and ((estimate == 'amplitude' or estimate == 'auto')):
         np.sqrt(psds, out=psds)
         psds *= scaling
         ylabel = '$\\frac{%s}{\\sqrt{Hz}}$' % unit
+
+    if not dB and (estimate == 'power'):
+        psds *= scaling * scaling
+        ylabel = '%s**2/Hz' % unit # Better to use latex and ^?
+
+    if dB and (estimate == 'amplitude'):
+        np.sqrt(psds, out=psds)
+        psds *= scaling
+        np.log10(psds, out=psds)
+        psds *= 10
+        ylabel = '$\\frac{%s}{\\sqrt{Hz}}\ (dB)$ ' % unit
```

Comment:
I would simplify the branching structure.

First

```
if 'auto':
    if dB:
         estimate='power'
    else:
         estimate='amplitude'
```
then it's obvious what auto means
after:

```
if estimate == "amplitude":
     np.sqrt(psds, out=psds)
     psds *= scaling
     ylabel = '$\\frac{%s}{\\sqrt{Hz}}$ ' % unit
else:
     ...

if dB:
    ylabel += "$\ (dB)$
```

Reply:
OK. Just pushed a version with simplified control flow.